controller/volume_controller_test.go

```diff
@@ -46,8 +46,15 @@ func setVolumeConditionWithoutTimestamp(originConditions map[string]longhorn.Con
 		Reason:  reason,
 		Message: message,
 	}
-	conditions[conditionType] = condition
-	return conditions
+
+	for i := range conditions {
+		if conditions[i].Type == conditionType {
+			conditions[i] = condition
+			return conditions
+		}
+	}
```

Comment:
Consider use the function `GetCondition`.

Reply:
Sure.

Reply:
The  data type of the condition is map[string]interface and not compatible with GetCondition's argument.
Do you have any idea for this?

Reply:
for example
```go
        i, _ := types.GetCondition(conditions, conditionType)
	if i != -1 {
		return conditions
	}
	return append(conditions, longhorn.Condition{
		Type:    conditionType,
		Status:  conditionValue,
		Reason:  reason,
		Message: message,
	})
```

Reply:
@jenting  Sorry. No prob for this. The problem is for the next comment.

Reply:
@jenting Sorry, the prob is for the comment in next comment (app/recurring_job.go).